### PR TITLE
File system tc enabling and bugfixs for ESP32

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
This PR is replaced with PR https://github.com/Samsung/TizenRT/pull/2846 
as it is not well organized.

this PR passed all filesystem testcase,  do below action
1) Add configs for enabling file system testcase.
2) Define smartfs partition device name for ESP32.
3) Change buf define method to fix testcase crash.
4) Check Null pointer in mtd_config function to fix testcase crash.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5bea721eb9834c1ebbeee497e8cba1319ce5cdae] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5bea721eb9834c1ebbeee497e8cba1319ce5cdae] - Code Rule Check (C++) OK.

File: build/configs/esp32_DevKitC/tash_wlan/defconfig [hunk 1]
```diff
@@ -198,16 +198,16 @@ CONFIG_ESPTOOLPY_AFTER_RESET=y
...
 #
-CONFIG_ARCH_USE_FLASH=y
 CONFIG_FLASH_PARTITION=y
 CONFIG_FLASH_MINOR=0
-CONFIG_FLASH_PART_SIZE="768,"
-CONFIG_FLASH_PART_TYPE="smartfs,"
-CONFIG_FLASH_PART_NAME="user,"
+CONFIG_FLASH_PART_SIZE="768,32,"
+CONFIG_FLASH_PART_TYPE="smartfs,config,"
+CONFIG_FLASH_PART_NAME="user,config"
```
Comment (yashwanth686007) [inline build/configs/esp32_DevKitC/tash_wlan/defconfig]: It's better to rename the partition name other than "config".

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 2]
```diff
@@ -1635,7 +1637,7 @@ static void tc_driver_mtd_config_ops(void)
 	int fd;
 	int ret;
 	struct config_data_s config;
-	char *buf = "test";
+	char buf[8] = "test";
```
Comment (jsdosa) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: i think this is better.
const char *buf = "test";

File: build/configs/esp_wrover_kit/tash_wlan/defconfig [hunk 3]
```diff
@@ -778,8 +781,18 @@ CONFIG_MTD=y
 CONFIG_MTD_PARTITION=y
 CONFIG_MTD_PARTITION_NAMES=y
 # CONFIG_MTD_PROGMEM is not set
-# CONFIG_MTD_FTL is not set
-# CONFIG_MTD_CONFIG is not set
+CONFIG_MTD_FTL=y
+
+#
+# MTD_FTL Configurations
```
Comment (SRCN-LiQin) [inline build/configs/esp_wrover_kit/tash_wlan/defconfig]: is it necessary to enable fs in tash_wlan config? maybe need another config dir?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 2 above)
Comment (ChinaTizenRT) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: as  ioctl cmd  "CFGDIOC_GETCONFIG"  will write this buf, 
if defined it as const, then normally SegmentFault
or will crash in TizenRT.

File: build/configs/esp32_DevKitC/tash_wlan/defconfig (hunk 1 above)
Comment (ChinaTizenRT) [reply build/configs/esp32_DevKitC/tash_wlan/defconfig]: rename it  to nvram as artik.

File: build/configs/esp_wrover_kit/tash_wlan/defconfig (hunk 3 above)
Comment (ChinaTizenRT) [reply build/configs/esp_wrover_kit/tash_wlan/defconfig]: yes, wifi utc and wm_test will save basic config info.
like current ap info to connect for station.

**Comment by seinfra** [comment]
Target : [9878983313900db9a669f1f7c3a7352c8a620192] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9878983313900db9a669f1f7c3a7352c8a620192] - Code Rule Check OK.

**Pull request merged**